src/preferences/dialog/dlgpreferences.cpp

```diff
@@ -406,9 +406,15 @@ void DlgPreferences::onShow() {
     int newY = m_geometry[1].toInt();
 
     const QList<QScreen*> screens = QGuiApplication::screens();
-    DEBUG_ASSERT(!screens.isEmpty());
-    newX = std::max(0, std::min(newX, screens.first()->geometry().width()- m_geometry[2].toInt()));
-    newY = std::max(0, std::min(newY, screens.first()->geometry().height() - m_geometry[3].toInt()));
+    QSize screenSpace;
+    VERIFY_OR_DEBUG_ASSERT(!screens.isEmpty()) {
+        qWarning() << "No screen was found, assuming screen size 800x600px.";
+        screenSpace = QSize(800, 600);
+    } else {
+        screenSpace = screens.first()->geometry().size();
```

Comment:
QList<Screen*>::first() is just one possible strategy for selecting the primary screen from this list. We can still hide it behind a one-line, inline utility function `Screen* getPrimaryScreen(QList<Screen*> screens)`, maybe in compatibility.h.

Reply:
Done in https://github.com/mixxxdj/mixxx/pull/2350/commits/8c1159e2e11186f6b289f7395c147e3d3976907b.